### PR TITLE
Support for remote repository, where all contents and manifests are fetched over HTTP(S) instead of locally manipulating blob storage

### PR DESCRIPTION
**Pull request by jkowalski**

### CODE REVIEW

**Comment by jkowalski** [comment]
(this is broken into smaller commits to facilitate reviewing and provide explanation, not guaranteed that they compile individually - they should, though)

File: repo/manifest/manifest_entry.go [hunk 1]
```diff
@@ -5,8 +5,8 @@ import "time"
 // EntryMetadata contains metadata about manifest item. Each manifest item has one or more labels
 // Including required "type" label.
 type EntryMetadata struct {
-	ID      ID
-	Length  int
-	Labels  map[string]string
-	ModTime time.Time
+	ID      ID                `json:"id"`
```
Comment (julio-lopez) [inline repo/manifest/manifest_entry.go:8]: Does this break existing (already serialized) manifests?

File: repo/local_config.go [hunk 2]
```diff
@@ -12,10 +12,12 @@ import (
 
 // LocalConfig is a configuration of Kopia stored in a configuration file.
 type LocalConfig struct {
-	Storage  blob.ConnectionInfo    `json:"storage"`
-	Caching  content.CachingOptions `json:"caching"`
-	Hostname string                 `json:"hostname"`
-	Username string                 `json:"username"`
+	APIServer *APIServerInfo `json:"apiServer"`
```
Comment (julio-lopez) [inline repo/local_config.go:15]: IIUC: a client can either directly use blob storage or go through the APIServer. Is that correct?

It makes sense to explicitly add a comment either to the struct or the fields about the intended usage.

File: repo/manifest/manifest_entry.go (hunk 1 above)
Comment (jkowalski) [reply repo/manifest/manifest_entry.go:8]: no, this struct was never serialized, with this change we'll start sending it over the wire.

File: repo/local_config.go (hunk 2 above)
Comment (jkowalski) [reply repo/local_config.go:15]: done.

**Pull request merged**